src/scenes/AcceleratorRouter/People/PersonProvider.tsx

```diff
@@ -18,37 +29,111 @@ const PersonProvider = ({ children }: Props) => {
   const dispatch = useAppDispatch();
   const snackbar = useSnackbar();
   const pathParams = useParams<{ userId: string }>();
-  const userId = Number(pathParams.userId);
+  const [userId, setUserId] = useState(Number(pathParams.userId || -1));
 
-  const [personData, setPersonData] = useState<PersonData>({ userId });
+  const getUserRequest = useAppSelector(selectUserRequest(userId));
+  const [getCategoriesRequestId, setGetCategoriesRequestId] = useState('');
+  const getCategoriesRequest = useAppSelector(selectUserDeliverableCategoriesGetRequest(getCategoriesRequestId));
 
-  const [user, setUser] = useState<User>();
-  const userRequest = useAppSelector(selectUserRequest(userId));
+  const [updateGlobalRolesRequestId, setUpdateGlobalRolesRequestId] = useState('');
+  const updateGlobalRolesRequest = useAppSelector(selectGlobalRolesUserUpdateRequest(updateGlobalRolesRequestId));
+  const [updateCategoriesRequestId, setUpdateCategoriesRequestId] = useState('');
+  const updateCategoriesRequest = useAppSelector(
+    selectUserDeliverableCategoriesUpdateRequest(updateCategoriesRequestId)
+  );
+  const [onUpdateSuccess, setOnUpdateSuccess] = useState<() => void>();
```

Comment:
I can't say that this is an anti-pattern, but it is somewhat abnormal. We generally don't set a function to state to be called later, instead we will watch a value and do things when the value changes to what we want. 

In combination with my other comment about the `PersonProvider`, I think what you should do here is create a `useUpdatePerson` hook that returns `{ update: (payload) -> void, success: boolean }`. Within the hook, if the update request fails, the hook can be responsible for sending the toast to the snackbar, and if the request is successful, it can update the `success` value within the hook's state so that downstream consumers know they can do specific things when the request is successful. 

In the case of its usage here, since you are navigating to a new view, any usages of the custom `useUpdatePerson` hook will be a new instantiation, so you don't have to worry about the `success` value "sticking" across views.

Reply:
I've made this change locally, but I'm not sure how best to get the provider to refresh its data after an edit succeeds. The problem is that I'm calling `useUpdatePerson()` from `EditView` and thus its success status isn't visible to the provider and can't trigger a refresh in the provider.

If I were to call `useUpdatePerson()` from the provider, I assume your last paragraph would then no longer apply since the hook would no longer be invoked from a new instantiation.

I can think of a couple ways to make it work but they feel like ugly hacks to me. I assume there's a clean way to do this.

Will update the PR with what I've done so you can see if it's just that I messed something up.

Reply:
I think in this case I would dispatch the fetch when the update resolves in the thunk, right above [this line](https://github.com/terraware/terraware-web/pull/2795/files#diff-316ee36fa58bf952f4168c6cdcea40825d639a60cff9cbae8a15f24cbc408494R27).

Reply:
Similar to [this](https://github.com/terraware/terraware-web/blob/main/src/redux/features/participants/participantsAsyncThunks.ts#L78)